### ISSUE TITLE
should we do something special for BN layers?

### ISSUE THREAD

**Issue by CinderellaRobaker** [kristpapadopoulos/keras_callbacks#4]
because in the swa (pytorch) [https://github.com/timgaripov/swa](url)
there is one step that use the data to update the BN layer? like
`utils.bn_update(loaders['train'], swa_model)`
is it necessary?

**Comment by simon-larsson** [comment]
Yes, it is necessary. Changing the weights in the end of training will offset the batch normalization.

**Comment by isaacgerg** [comment]
I believe on way to fix this is to freeze all the layers but the  batchnorm layers and do a few batches of the forward pass (training).

**Comment by simon-larsson** [comment]
You don't need to freeze all layers. You can just run an epoch with learning rate set to zero.

**Comment by isaacgerg** [comment]
Do what @simon-larsson says, its a much more elegant solution.

**Comment by CinderellaRobaker** [comment]
您好~这里是罗壮的邮箱，我已收到您的来信，会尽快给您回复的，请稍作等待喔~
